libp2p/protocols/pubsub/rpc/messages.nim

```diff
@@ -9,14 +9,15 @@
 
 import options, sequtils
 import ../../../utility
+import ../../../peer
 
 type
     SubOpts* = object
       subscribe*: bool
       topic*: string
 
     Message* = object
-      fromPeer*: seq[byte]
+      fromPeer*: PeerId
```

Comment:
this is supposed to be bytes by spec.. is it compatible?
Also 1.1 specifies a peerInfo on its own

Reply:
well, the issue here was basically that the field is used as a peer id but because it had the weak type, it would introduce the potential for exceptions at every place it's parsed + it's kind of wasteful to parse over and over again - this is just a hotfix to ensure parsing is done in one place (when message arrives)

Reply:
This is wrong, your changing the rpc and it will break other implementation, we can filter out the peerid I'm the observers.

Reply:
on a second though, if peerid bytes are what ultimately hits the wire then we're fine and it's a good change in fact.

Reply:
:smile: generally, this could be done with the other fields too, but they're not as vulnerable as this one..

Reply:
yeah, all for it :), this needs cleanup either way.